Drainage system 2

javierNadal opened the pull request:
## DrainageSystem-2

Drainage around Sammalammintie Road2 intersection. Ditchs and culvert represented by 3D polylines (Axis).

## Covered topics

*tick one/many*

- [ ] project setup
- [ ] georeferencing
- [ ] alignment semantics
- [ ] alignment geometry
- [ ] linear placement
- [x] project breakdown structure / spatial structure
- [ ] terrain
- [ ] triangulated irregular network
- [ ] open cross profile
- [ ] sectioned surface & solid
- [ ] pavement & course
- [ ] surface feature
- [ ] earthworks
- [ ] geotechnics
- [ ] longitudinal products (barriers, guardrail, ...)
- [ ] interchange
- [ ] structural
- [x] drainage
- [ ] signage
- [ ] road furniture
- [ ] rail furniture
- [ ] ports & waterways furniture
- [ ] bridge furniture

javierNadal committed:
Last alignment changes pending (Meeting Feb 10th)

javierNadal committed:
Alignment-Aplitop-1 RC3

javierNadal committed:
readme corrections

javierNadal committed:
More readme corrections

javierNadal committed:
Merge pull request #1 from bSI-InfraRoom/main

Updating changes

javierNadal committed:
Updated with latest schema (12 March 2021)

javierNadal committed:
Wrong file

javierNadal committed:
Merge pull request #2 from bSI-InfraRoom/main

Updating

javierNadal committed:
Alignment with 2D representation

javierNadal committed:
Merge pull request #3 from bSI-InfraRoom/main

Updating

javierNadal committed:
DrainageSystem-2 with Polyline representation

javierNadal committed:
Merge branch 'DrainageSystem-2' of https://github.com/javierNadal/IFC-infra-unit-test into DrainageSystem-2

pjanck commented:
There is:

- either a new readme for the drainage example missing
- or an existing readme not yet amended to reference to the drainage IFC file

larswik commented:
Please check the following:

- The physical elements are never contained in a spatial structure element Simplest option would be to contain all in IfcSite; Alternatively, ditches named 8 and 204 in Road2, others in Sammalamminitie (or in its parts)

- Physical elements (or their geometric representations), need to consider the geometric representation context and especially the IfcMapConversion.Northings/Eastings (needs to be subtracted). Also consider using separate local placements 

- The topology needs to be checked. Three pipe segments are connected to the distribution chamber at one end but none are connected at the other end. This is not an error at ditch ends (elements where they would connect not modeled in this UT), but the culvert should be connected at both ends (see next).

- Obviously because the element and its ports where they would connect is missing (there’s only one IfcDistributionChamberElement).

- Drainage system is never declared on project level. Use IfcRelDeclares.

- The use of SOURCE/SINK seem to be inconsistent with regards to relating/related in IfcRelConnectsPorts and e.g. the one  IfcDistributionChamberElement has all its ports of type SINK. 

- Distribution chamber element has no geometric representation – it might be useful to take have a geometric representation. It could be in this case a short line where you could place one port on each end and one in the middle. 
Note: also the ports on pipe segments should have local placements.

![image](https://user-images.githubusercontent.com/25506404/114673448-7d26ee00-9d06-11eb-856a-32fe01675f57.png)

javierNadal committed:
Updated ifc file

javierNadal committed:
Added IfcRelContainedInSpatialStructure and IfcRelReferencedInSpatialStructure to between drainage elements and roads

larswik committed:
Small adjustments

Renaming ifc-file (think @pjanck prefers dash "-" instead of underscore "_")
Added ifc-file to the supporting files listing.

larswik commented:
I think that the ifc-file from @javierNadal is ok. I have sent to Juha and await his response.
I think that I have also fixed the readme issue from @pjanck (adding the ifc file to the supporting files listing). 
When Juha gives a green light, I think that I can merge.

larswik committed:
Update DrainageSystem-2.IFC

Juha updated the SOURCE vs SINK.
